src/OrchardCore.Modules/OrchardCore.Queries/Sql/SqlQuerySource.cs

```diff
@@ -58,15 +59,17 @@ public async Task<object> ExecuteQueryAsync(Query query, IDictionary<string, obj
                         
             if (sqlQuery.ReturnDocuments)
             {
-                IEnumerable<int> documentIds;
+                int[] documentIds;
 
                 using (connection)
                 {
                     connection.Open();
-                    documentIds = await connection.QueryAsync<int>(rawQuery, parameters);
+                    documentIds = (await connection.QueryAsync<int>(rawQuery, parameters)).ToArray();
                 }
+                
+                var documents = await _session.GetAsync<object>(documentIds);
 
-                return await _session.GetAsync<object>(documentIds.ToArray());
+                return documents.OrderBy(d => Array.IndexOf(documentIds, ((dynamic)d).Id));
```

Comment:
I would fix it in yessql, here: https://github.com/sebastienros/yessql/blob/dev/src/YesSql.Core/Session.cs#L322

I can do it. I already ensure it keeps the same order in the other methods.

Reply:
I just submitted a PR to yessql for the fix since I had already started on it: https://github.com/sebastienros/yessql/pull/138